build-tools/packages/build-tools/src/type-test-generator/main.ts

```diff
@@ -4,96 +4,51 @@
  */
 
 import { readJsonSync } from "fs-extra";
-import { existsSync, mkdirSync, rmSync, writeFileSync } from "node:fs";
+import { mkdirSync, writeFileSync } from "node:fs";
 import path from "node:path";
-import { Project, SourceFile } from "ts-morph";
-import { BrokenCompatTypes } from "../common/fluidRepo";
-import { PackageJson } from "../common/npmPackage";
-import { buildTestCase, TestCaseTypeData } from "../typeValidator/testGeneration";
-import { getFullTypeName, getNodeTypeData, TypeData } from "../typeValidator/typeData";
 import { typeOnly } from "./compatibility";
+import {
+	ensureDevDependencyExists,
+	getTypeRollupPathFromExtractorConfig,
+	getTypeDefinitionFilePath,
+	typeDataFromFile,
+	generateCompatibilityTestCases,
+	prepareAndSkipTestGenerationIfDisabled,
+	initializeProjectsAndLoadFiles,
+} from "./typeTestUtils";
+import { PackageJson } from "../common/npmPackage";
 
 // Do not check that file exists before opening:
 // Doing so is a time of use vs time of check issue so opening the file could fail anyway.
 // Do not catch error from opening file since the default behavior is fine (exits process with error showing useful message)
 const packageObject: PackageJson = readJsonSync("package.json");
-
-const testPath = `./src/test/types`;
-// remove scope if it exists
-const unscopedName = path.basename(packageObject.name);
-
-const fileBaseName = unscopedName
-	.split("-")
-	.map((p) => p[0].toUpperCase() + p.substring(1))
-	.join("");
-const filePath = `${testPath}/validate${fileBaseName}Previous.generated.ts`;
-
-if (packageObject.typeValidation?.disabled) {
-	console.log("skipping type test generation because they are disabled in package.json");
-	// force means to ignore the error if the file does not exist.
-	rmSync(filePath, { force: true });
-	process.exit(0);
-}
-
 const previousPackageName = `${packageObject.name}-previous`;
+const previousBasePath = path.join("node_modules", previousPackageName);
 
-{
-	// Information about the previous package from the package.json is not needed,
-	// but error if its missing since it's nice to separate errors for the dep missing here vs not installed.
-	const previousDep = packageObject?.devDependencies?.[previousPackageName];
-	if (typeof previousDep !== "string") {
-		throw new Error(`Did not find devDependency ${previousPackageName} in package.json`);
-	}
+try {
+	ensureDevDependencyExists(packageObject, previousPackageName);
+} catch (error) {
+	console.error(error);
 }
```

Comment:
I don't think we want the try catch here? It would log the error but then continue executing, which is not what happened before. Letting the error propagate would maintain the current behavior.